sycl/source/detail/async_alloc.cpp

```diff
@@ -135,9 +146,15 @@ async_malloc_from_pool(const sycl::queue &q, size_t size,
 }
 
 __SYCL_EXPORT void async_free(sycl::handler &h, void *ptr) {
-  h.throwIfGraphAssociated<
-      ext::oneapi::experimental::detail::UnsupportedGraphFeatures::
-          sycl_ext_oneapi_async_alloc>();
+  if (auto Graph = h.getCommandGraph(); Graph) {
+    // Check if the pointer to be freed has an associated allocation node, and
+    // error if not
+    if (!Graph->getMemPool().hasAllocation(ptr)) {
```

Comment:
Can we add a section to the design doc on the async malloc/free impl. Doesn't need to be a lot, but could note things like free being a no-op as we don't do memory reallocation yet.

Reply:
I've added some wording for these now, thanks!